File: output/CodeGen/avatar/Python/Java_repair/codeforces_581_B.java
import java.util.*;

public class codeforces_581_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int[] nums = scanner.nextInt().split();
        int k = Integer.parseInt(scanner.next());
        int[] ans = maxSlidingWindow(nums, k);
        System.out.println(ans);
    }
    public static int[] maxSlidingWindow(int[] nums, int k) {
        int[] ans = new int[nums.length - k + 1];
        int mx = nums[0];
        for (int i = 0; i < nums.length - k + 1; i++) {
            ans[i] = Math.max(0, mx - nums[i]);
            if (nums[i] > mx) {
                mx = nums[i];
            }
        }
        return ans;
    }
}

File: output/CodeGen/avatar/Java/Python/codeforces_581_B.py
import java.util.* ; import java.io.* ; 

def main ():
    in = raw_input()
    n = int(in)
    a = map(int,raw_input().split())
    dp = [ -1 for i in range(n+1)]
    dp[n-1] = a[n-1]
    for i in range(n-1,-1,-1):
        dp[i] = max(dp[i+1],a[i])
    for i in range(n):
        if a[i] > dp[i+1]:
            print 0,
        else:
            print dp[i+1]-a[i]+1,
    print

if __name__ == "__main__":
    main()